src/ripple/protocol/SField.h

```diff
@@ -165,18 +165,23 @@ class SField
     {
         return getField (field_code (type, value));
     }
+
     static const SField& getField (SerializedTypeID type, int value)
     {
         return getField (field_code (type, value));
     }
 
-    std::string getName () const;
+    std::string const& getName () const
+    {
+        return fieldName;
+    }
+
     bool hasName () const
     {
-        return !fieldName.empty ();
+        return fieldNum > 0;
```

Comment:
This can return `true` with the `SField::SField (int fc)` ctor.

Reply:
Good catch. That should be "fieldCode". It only affects an edge case (where we encounter a field we don't understand).